quinn-proto/src/connection/mod.rs

```diff
@@ -1529,7 +1529,7 @@ impl Connection {
                 self.stats.frame_tx.ping += 1;
 
                 // If supported by the peer, we want no delays to the probe's ACK
-                if self.peer_supports_ack_frequency() {
+                if self.peer_supports_ack_frequency() && space_id == SpaceId::Data {
```

Comment:
oh no! we should not MTUD probe in the handshake! I added a bug when I wrote this, and this is the wrong fix.